applications/nrf_desktop/doc/hid_provider_mouse.rst

```diff
@@ -42,90 +56,52 @@ See the utility's documentation for details.
 Implementation details
 **********************
 
-On initialization, the module announces its presence by sending a :c:struct:`hid_report_provider_event` event with appropriate report ID and implementation of callbacks from the HID report provider API (:c:struct:`hid_report_provider_api`).
-Separate :c:struct:`hid_report_provider_event` events are sent for each report ID.
-The module supports mouse reports from the report and boot protocols.
-You can enable the boot protocol support using the :ref:`CONFIG_DESKTOP_HID_BOOT_INTERFACE_MOUSE <config_desktop_app_options>` Kconfig option.
-The :ref:`nrf_desktop_hid_state` receives the events and fills them with its own implementation of callbacks from the HID state API (:c:struct:`hid_state_api`).
-After that process, the modules can communicate by callbacks.
-The module also subscribes to the :c:struct:`button_event`, :c:struct:`wheel_event` and :c:struct:`motion_event` events to get information about the button presses, wheel scrolls, and motion events.
-The module sends :c:struct:`hid_report_event` events to an appropriate subscriber when it is requested by the :ref:`nrf_desktop_hid_state`.
-
-.. note::
-    The HID report formatting function must work according to the HID report descriptor (``hid_report_desc``).
-    The source file containing the descriptor is provided by the :ref:`CONFIG_DESKTOP_HID_REPORT_DESC <config_desktop_app_options>` Kconfig option.
+The module is used by :ref:`nrf_desktop_hid_state` as a HID input report provider for the HID mouse input report and HID boot mouse input report.
+The module registers two separate HID report providers to handle both input reports.
+On initialization, the module submits the :c:struct:`hid_report_provider_event` events to establish two-way callbacks between the |hid_state| and the HID report providers.
 
-Linking input data with the right HID report
-============================================
+Handling user input
+===================
 
-Out of all available input data types, the following types are collected by this module:
+The module handles the following types of user input:
 
 * `Relative value data`_ (axes)
+  This type of input data is related to the pointer coordinates and the wheel rotation.
+  Both :c:struct:`motion_event` and :c:struct:`wheel_event` are sources of this type of data.
 * `Absolute value data`_ (buttons)
+  This type of input data is related to the state of pressed buttons.
+  The :c:struct:`button_event` is the source of this type of data.
 
-Both types are stored in the :c:struct:`report_data` structure.
+Both input types are stored internally in the :c:struct:`report_data` structure.
```

Comment:
```suggestion
Both input types are stored internally in the :c:struct:`report_data` structure.
```
The report_data struct doesn't link. so, maybe in double backticks?

Reply:
It's an internal structure. I can change to double backticks